crates/cac_client/src/interface.rs

```diff
@@ -102,15 +102,40 @@ pub extern "C" fn cac_new_client(
     tenant: *const c_char,
     update_frequency: c_ulong,
     hostname: *const c_char,
+    cache_max_capacity: *const c_char,
+    cache_ttl: *const c_ulong,
+    cache_tti: *const c_ulong,
 ) -> c_int {
     let duration = Duration::new(update_frequency, 0);
     let tenant = unwrap_safe!(cstring_to_rstring(tenant), return 1);
     let hostname = unwrap_safe!(cstring_to_rstring(hostname), return 1);
 
+    let cache_max_capacity = if !cache_max_capacity.is_null() {
+        Some(unsafe { *cache_max_capacity as u64 })
```

Comment:
this should be in `unwrap_safe!`

Reply:
@ayushjain17 unwrap_safe! is for unwraping and handling error ,
but dereferencing a pointer is unsafe function , so I need unsafe block